raiden/network/transport/udp/udp_transport.py

```diff
@@ -204,14 +211,31 @@ def start(
             self.init_queue_for(queue_identifier, encoded_queue)
 
         self.server.start()
+        super().start()
+
+    def run(self):
+        """ Runnable main method, perform wait on long-running subtasks """
+        try:
+            self.stop_event.wait()
+        except gevent.GreenletExit:  # killed without exception
+            self.stop_event.set()
+            gevent.killall(self.greenlets)  # kill children
+            raise  # re-raise to keep killed status
+        except Exception:
+            self.stop()  # ensure cleanup and wait on subtasks
+            raise
+
+    def stop(self):
+        if self.stop_event.ready():
+            return  # double call, happens on normal stop, ignore
```

Comment:
Yes, this simple return was from a left-over change where `stop` could be called twice (once for graceful exit, other on `run` teardown), but since `run` now only calls `stop` when raising, this shouldn't be the case anymore and can successfuly raise if called on non-started Runnable.